pkg/oauthserver/help_test.go

```diff
@@ -0,0 +1,97 @@
+package oauthserver_test
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/openshift/oauth-server/pkg/audit"
+	apiaudit "k8s.io/apiserver/pkg/apis/audit"
+	kaudit "k8s.io/apiserver/pkg/audit"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+func withKeyValues(values ...[2]string) string {
+	const (
+		key   = 0
+		value = 1
+	)
+
+	form := url.Values{}
+
+	for _, tuple := range values {
+		form.Add(tuple[key], tuple[value])
+	}
+
+	return form.Encode()
+}
+
+func withKeyValue(key, value string) [2]string {
+	return [2]string{key, value}
+}
+
+func WithAuth(req *http.Request, username, password string) {
+	req.Header.Add(
+		"Authorization",
+		"Basic "+base64.StdEncoding.EncodeToString([]byte(username+":"+password)),
+	)
+}
+
+func withDecisionAnnotation(want string) func(map[string]string) error {
+	return func(ano map[string]string) error {
+		if have, ok := ano[audit.DecisionAnnotation]; !ok || want != have {
+			return fmt.Errorf("want: %s for decision, have: %s for decision", want, have)
+		}
+
+		return nil
+	}
+}
+
+func withUsernameAnnotation(want string) func(map[string]string) error {
+	return func(ano map[string]string) error {
+		have, ok := ano[audit.UsernameAnnotation]
+		valueExpected := len(want) > 0
+
+		if !ok && valueExpected || want != have {
+			return fmt.Errorf(
+				"want: '%s' for username, have: '%s' for username and ok is %t",
+				want, have, ok,
+			)
+		}
+
+		return nil
+	}
+}
+
+type annotationChecker func(map[string]string) error
+
+func annotationChecks(checks ...annotationChecker) []annotationChecker {
+	return checks
+}
+
+func verifyAnnotations(req *http.Request, checker []annotationChecker) error {
+	ev, err := kaudit.NewEventFromRequest(
+		req, time.Now(),
+		apiaudit.LevelRequestResponse,
+		&authorizer.AttributesRecord{},
+	)
+
+	if err != nil {
+		panic(err)
```

Comment:
nit: you can pass a `t *testing.T` instead and prevent panicing here.

Reply:
Done, and everywhere else I did that